File: cloud-storage-spring-java-example/src/main/java/com/github/sevtech/cloud/storage/spring/java/example/TestApplication.java
package com.github.sevtech.cloud.storage.spring.java.example;

import com.github.sevtech.cloud.storage.spring.bean.DeleteFileRequest;
import com.github.sevtech.cloud.storage.spring.bean.GetFileRequest;
import com.github.sevtech.cloud.storage.spring.bean.UploadFileRequest;
import com.github.sevtech.cloud.storage.spring.bean.UploadFileResponse;
import com.github.sevtech.cloud.storage.spring.service.StorageService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.scheduling.annotation.EnableAsync;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.multipart.MultipartFile;

import java.io.ByteArrayInputStream;
import java.io.IOException;

@RestController
@RequestMapping
@EnableAsync
@SpringBootApplication
public class TestApplication {

    public static void main(String[] args) {
        SpringApplication.run(TestApplication.class, args);
    }

    @Autowired
    private StorageService storageService;

    @PostMapping("/files")
    public UploadFileResponse uploadFile(@RequestBody MultipartFile file, @RequestParam String name, @RequestParam String folder) throws IOException {
        return storageService.uploadFile(UploadFileRequest.builder().stream(new ByteArrayInputStream(file.getBytes())).folder(folder).name(name).contentType(file.getContentType()).build());
    }

    @GetMapping("/files")
    public byte[] getFile(@RequestParam String name) {
        return storageService.getFile(GetFileRequest.builder().path(name).build()).getContent();
    }

    @DeleteMapping("/files")
    public int deleteFile(@RequestParam String name) {
        return storageService.deleteFile(DeleteFileRequest.builder().path(name).build()).getStatus();
    }

}

File: cloud-storage-spring-kotlin-example/src/main/kotlin/com/github/sevtech/localstack/kotlin/example/TestApplication.kt
package com.github.sevtech.localstack.kotlin.example

import com.github.sevtech.cloud.storage.spring.model.DeleteFileRequest
import com.github.sevtech.cloud.storage.spring.model.GetFileRequest
import com.github.sevtech.cloud.storage.spring.model.UploadFileRequest
import com.github.sevtech.cloud.storage.spring.model.UploadFileResponse
import com.github.sevtech.cloud.storage.spring.service.StorageService
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.scheduling.annotation.EnableAsync
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile
import java.io.ByteArrayInputStream
import java.io.IOException

@EnableAsync
@SpringBootApplication
class TestApplication

fun main(args: Array<String>) {
    SpringApplication.run(TestApplication::class.java, *args)
}

@RestController
@RequestMapping("/")
class MediaController(private val storageService: StorageService) {


    @PostMapping("/files")
    @Throws(IOException::class)
    fun uploadFile(@RequestBody file: MultipartFile, @RequestParam name: String, @RequestParam folder: String): UploadFileResponse {
        return storageService.uploadFile(UploadFileRequest(stream = ByteArrayInputStream(file.bytes), folder = folder, name = name, contentType = file.contentType!!))
    }

    @GetMapping("/files")
    @Throws(IOException::class)
    fun getFile(@RequestParam name: String): ByteArray? {
        return storageService.getFile(GetFileRequest(path = name)).content
    }

    @DeleteMapping("/files")
    @Throws(IOException::class)
    fun deleteFile(@RequestParam name: String): Int {
        return storageService.deleteFile(DeleteFileRequest(path = name)).status
    }

}


